ecosystem/sf-indexer/aptos-substreams/src/token_converter.rs

```diff
@@ -0,0 +1,369 @@
+                    txn_version, value
+                ))?
+                .context(format!(
+                    "version {} failed! failed to parse royalty_points_numerator {:?}",
+                    txn_version, value["royalty"]["royalty_points_numerator"]
+                ))?,
+            royalty_points_denominator: value["royalty"]["royalty_points_denominator"]
+                .as_str()
+                .map(|s| s.parse::<u64>())
+                .context(format!(
+                    "version {} failed! royalty.royalty_points_denominator missing {:?}",
+                    txn_version, value
+                ))?
+                .context(format!(
+                    "version {} failed! failed to parse royalty_points_denominator {:?}",
+                    txn_version, value["royalty"]["royalty_points_denominator"]
+                ))?,
+            maximum_mutable: value["mutability_config"]["maximum"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.maximum missing {:?}",
+                    txn_version, value
+                ))?,
+            uri_mutable: value["mutability_config"]["uri"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.uri missing {:?}",
+                    txn_version, value
+                ))?,
+            description_mutable: value["mutability_config"]["description"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.description missing {:?}",
+                    txn_version, value
+                ))?,
+            properties_mutable: value["mutability_config"]["properties"].as_bool().context(
+                format!(
+                    "version {} failed! mutability_config.properties missing {:?}",
+                    txn_version, value
+                ),
+            )?,
+            royalty_mutable: value["mutability_config"]["royalty"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.royalty missing {:?}",
+                    txn_version, value
+                ))?,
+            default_properties: value["default_properties"]
+                .as_object()
+                .map(|s| serde_json::to_string(s))
+                .context(format!(
+                    "version {} failed! default_properties missing from token data {:?}",
+                    txn_version, value
+                ))?
+                .context(format!(
+                    "version {} failed! failed to stringify default_properties {:?}",
+                    txn_version, value["default_properties"]
+                ))?,
+        }));
+    }
+    Ok(None)
+}
+
+pub fn get_collection_data(
+    table_item: &TableItemOutput,
+    txn_version: u64,
+    table_handle_to_owner: &HashMap<String, (String, String)>,
```

Comment:
You should use a type to describe the three strings, for someone who looks at this, it's wondering what's the 2 strings at the end?  Especially since you throw away the second one a few lines below

Reply:
Done.